src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -518,8 +518,8 @@ public bool IsDotNetCore
                 // The .NET core bootstrap exe is a generated native binary that loads
                 // its corresponding .NET core application entry point dll.
                 !this.IsDotNetCore
-                    && this.CodeViewDebugDirectoryData.Path != null
-                    && this.CodeViewDebugDirectoryData.Path.EndsWith("apphost.pdb", StringComparison.OrdinalIgnoreCase);
+                    && (this.CodeViewDebugDirectoryData.Path?.EndsWith("apphost.pdb", StringComparison.OrdinalIgnoreCase) == true
+                        || this.CodeViewDebugDirectoryData.Path?.EndsWith("singlefilehost.pdb", StringComparison.OrdinalIgnoreCase) == true);

```

Comment:
(this.CodeViewDebugDirectoryData.Path?.EndsWith("apphost.pdb", StringComparison.OrdinalIgnoreCase) || 
                        this.CodeViewDebugDirectoryData.Path?.EndsWith("singlefilehost.pdb", StringComparison.OrdinalIgnoreCase));